test/cxx/ServerKit/HttpServerTest.cpp

```diff
@@ -725,50 +725,76 @@ namespace tut {
 
 		connectToServer();
 		sendRequest(
-			"POST / HTTP/1.1\r\n"
-			"Host: whatever\r\n"
-			"Transfer-Encoding: ,chunked\r\n"
-			"Content-Length: 5\r\n"
-			"\r\n"
-			"0\r\n"
-			"\r\n");
+					"POST / HTTP/1.1\r\n"
+					"Host: whatever\r\n"
+					"Transfer-Encoding: ,chunked\r\n"
+					"Content-Length: 5\r\n"
+					"\r\n"
+					"0\r\n"
+					"\r\n");
 		string response = readAll(fd, 1024).first;
-		ensure(containsSubstring(response,
-			"HTTP/1.0 400 Bad Request\r\n"
-			"Status: 400 Bad Request\r\n"
-			"Content-Type: text/html; charset=UTF-8\r\n"));
-		ensure(containsSubstring(response,
-			"Connection: close\r\n"
-			"Content-Length: 79\r\n"
-			"cache-control: no-cache, no-store, must-revalidate\r\n"
-			"\r\n"
-			"Bad request (request may not contain both Content-Length and Transfer-Encoding)"));
+		ensure(startsWith(response,
+					  "HTTP/1.0 400 Bad Request\r\n"
+					  "Status: 400 Bad Request\r\n"
+					  "Content-Type: text/html; charset=UTF-8\r\n"));
+		ensure(endsWith(response,
+					  "Connection: close\r\n"
+					  "Content-Length: 42\r\n"
+					  "cache-control: no-cache, no-store, must-revalidate\r\n"
+					  "\r\n"
+					  "invalid character in content-length header"));
+		ensure_equals(response.size(),265u);
```

Comment:
I don't think we need to be _this_ strict with checking the response. It doesn't add any value to the purpose of the test.

Reply:
The test has to make sure there's no response other than the one it expects. Because setting the date isn't implemented for error responses I can't check the exact value of the entire response, so I just check the start and end and make sure the length matches the two ends plus the date line. It's technically under-specified but I didn't want to add the date-overriding to the test harness.

I used to just check the result using 'contains' but that wound up with false negatives.